src/rustllvm/PassWrapper.cpp

```diff
@@ -1496,6 +1498,42 @@ LLVMRustParseBitcodeForLTO(LLVMContextRef Context,
   return wrap(std::move(*SrcOrError).release());
 }
 
+// njn: update comment
+// This is what we used to parse upstream bitcode for actual ThinLTO
+// processing.  We'll call this once per module optimized through ThinLTO, and
+// it'll be called concurrently on many threads.
+extern "C" bool  // njn: return type is wrong
+LLVMRustExtractBitcodeFromObject(//LLVMContextRef Context,  // njn: Context not needed?
+                                 const char *data,
+                                 size_t len,
+                                 const char *identifier,
+                                 const char **out_data,
+                                 size_t *out_len) {
+  StringRef Data(data, len);
+  MemoryBufferRef Buffer(Data, identifier);
+  //unwrap(Context)->enableDebugTypeODRUniquing();  // njn: ?
+
+  Expected<std::unique_ptr<object::ObjectFile>> ObjectFileOrError =
+    object::ObjectFile::createObjectFile(Buffer);
+  if (!ObjectFileOrError) {
+    fprintf(stderr, "createObjectFile failed\n");
+    LLVMRustSetLastError(toString(ObjectFileOrError.takeError()).c_str());
+    return false;
+  }
+
+  Expected<MemoryBufferRef> BitcodeOrError =
+    object::IRObjectFile::findBitcodeInObject(**ObjectFileOrError);
+  if (!BitcodeOrError) {
+    fprintf(stderr, "findBitcodeInObject failed\n");
+    LLVMRustSetLastError(toString(BitcodeOrError.takeError()).c_str());
+    return false;
+  }
+
+  *out_data = BitcodeOrError->getBufferStart();
```

Comment:
FWIW this might be easier if the pointer is returned instead of true/false, and that way you wouldn't need `&mut *mut`.

Reply:
True, I will do that.